singularity-opac/neutrinos/mean_opacity_neutrinos.hpp

```diff
@@ -46,6 +46,24 @@ class MeanOpacity {
               const int NRho, const Real lTMin, const Real lTMax, const int NT,
               const Real YeMin, const Real YeMax, const int NYe,
               Real *lambda = nullptr) {
+    MeanOpacityImpl<Opacity, true>(opac, lRhoMin, lRhoMax, NRho, lTMin, lTMax, NT, YeMin, YeMax,
+                NYe, -1., -1., 100, lambda);
+  }
+
+  template <typename Opacity>
+  MeanOpacity(const Opacity &opac, const Real lRhoMin, const Real lRhoMax,
+              const int NRho, const Real lTMin, const Real lTMax, const int NT,
+              const Real YeMin, const Real YeMax, const int NYe, Real lNuMin,
+              Real lNuMax, const int NNu, Real *lambda = nullptr) {
+    MeanOpacityImpl<Opacity, false>(opac, lRhoMin, lRhoMax, NRho, lTMin, lTMax, NT, YeMin, YeMax,
+                NYe, lNuMin, lNuMax, NNu, lambda);
+  }
+
+  template <typename Opacity, bool AUTOFREQ>
+  MeanOpacityImpl(const Opacity &opac, const Real lRhoMin, const Real lRhoMax,
+              const int NRho, const Real lTMin, const Real lTMax, const int NT,
+              const Real YeMin, const Real YeMax, const int NYe, Real lNuMin,
+              Real lNuMax, const int NNu, Real *lambda = nullptr) {
```

Comment:
Does this actually work? I think the constructor has to be named exactly the same as the class. Maybe you need to go to tag dispatch or something?